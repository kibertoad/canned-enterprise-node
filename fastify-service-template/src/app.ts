import http from 'http'
import pino from 'pino'
import fastify, { FastifyInstance } from 'fastify'
import fastifyHelmet from 'fastify-helmet'
import fastifyCors from 'fastify-cors'
import middie from 'middie'

const healthCheck = require('fastify-healthcheck')

export async function getApp(): Promise<
  FastifyInstance<http.Server, http.IncomingMessage, http.ServerResponse, pino.Logger>
> {
  const app = fastify<http.Server, http.IncomingMessage, http.ServerResponse, pino.Logger>({
    logger: true
  })

  await app.register(middie)
  app.register(fastifyHelmet)
  app.register(healthCheck)

  app.register(fastifyCors, {
    methods: ['GET', 'POST'],
    credentials: false,
    maxAge: 86400,
    origin: []
  })

  try {
    await app.ready()
  } catch (err) {
    app.log.error('Error while initializing app: ', err)
    throw err
  }

  return app
}
