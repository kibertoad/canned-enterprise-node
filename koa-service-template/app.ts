import Koa, { Middleware } from 'koa'
import Router from 'koa-router'
import Application from 'koa'

function getMiddleware(): readonly Middleware[] {
  return []
}

function getRouters(): readonly Router[] {
  return []
}

function getApp(): Application {
  const app = new Koa()

  const appMiddleware = getMiddleware()
  appMiddleware.forEach((middlewareEntry: Middleware) => {
    app.use(middlewareEntry)
  })

  const appRouters = getRouters()
  appRouters.forEach((router: Router) => {
    app.use(router.routes()).use(router.allowedMethods())
  })

  return app
}

module.exports = {
  getApp,
}
