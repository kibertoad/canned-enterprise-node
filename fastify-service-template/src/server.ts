import { getApp } from './app'
import { getConfig } from './infrastructure/config'

const config = getConfig()

async function start() {
  const app = await getApp()

  try {
    await app.listen(config.app.port, config.app.bindAddress)
  } catch (err: any) {
    app.log.error(err.message)
    process.exit(1)
  }
}

start()
