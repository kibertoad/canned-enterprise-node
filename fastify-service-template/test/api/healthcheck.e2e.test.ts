import { getApp } from '../../src/app'
import fastify from 'fastify'

describe('healthcheck', () => {
  let app: fastify.FastifyInstance
  beforeEach(async () => {
    app = await getApp()
  })

  afterEach(() => {
    return app.close()
  })

  it('Returns health check information', async () => {
    // @ts-ignore
    const response = await app.inject().get('/health')

    expect(response.statusCode).toEqual(200)
    expect(response.json()).toMatchSnapshot()
  })
})
