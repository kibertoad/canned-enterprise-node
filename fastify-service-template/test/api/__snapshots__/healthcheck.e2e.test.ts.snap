// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`healthcheck Returns health check information 1`] = `
Object {
  "status": "ok",
  "statusCode": 200,
}
`;
